lib/util/cmcd_manager.js

```diff
@@ -55,6 +58,30 @@ shaka.util.CmcdManager = class {
      * @private {boolean}
      */
     this.starved_ = false;
+
+    /**
+     * @private {shaka.extern.RequestFilter}
+     */
+    this.requestFilter_ = (type, request) => {
```

Comment:
Why would the creator of the HEAD request not call into CMCD manager instead?  Could you add a comment?

Reply:
Using a request filter keeps the CMCD logic contained to the manager class. Otherwise the `cmcdManager` instance needs to be passed around to every place a `HEAD` request is made, which would deeply ingrain the feature in the code base. For example, it would need to be passed to:
- The static `getMimeType` and `getFactory` functions of `Manifest` parser
- The instances of all the format specific manifest parsers.